src/fft.js

```diff
@@ -91,6 +91,17 @@ define(function (require) {
    *  </code></div>
    */
   p5.FFT = function(smoothing, bins) {
+    Object.defineProperty(this, 'bins', {
+      get: function () {
+        return this.analyser.fftSize / 2;
```

Comment:
I think we need to define `this.analyser` before we do `Object.defineProperty`

From there, we're good to go. And maybe we can do something like this

```
    var BINS = bins || 1024;
    var SMOOTHING = smoothing || 0.8;
    
    // create analyser
    this.input = this.analyser = p5sound.audiocontext.createAnalyser();

    // define properties
    Object.defineProperties(this, {
      'bins': {
        get: function() {
          return this.analyser.fftSize / 2;
        },
        set: function(b) {
          this.analyser.fftSize = b * 2;
        },
        configurable: true,
        enumerable: true
      },
      'smoothing': {
        get: function() {
          return this.analyser.smoothing;
        },
        set: function(s) {
          this.analyser.smoothingTimeConstant = s;
        },
        configurable: true,
        enumerable: true
      }
    });

    // now we can use the "setters"
    // to set the analyser and the p5.FFT props at the same time
    this.smoothing = SMOOTHING;
    this.bins = BINS;
```

Feel free to make that change or let me know if I should do it after merging these PR's. Thanks for inspiring these improvements, @TheBrenny !